iceoryx_posh/test/moduletests/test_roudi_iceoyx_roudi_memory_manager.cpp

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) 2021 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "test.hpp"
+
+#include "iceoryx_posh/roudi/memory/iceoryx_roudi_memory_manager.hpp"
+
+using namespace ::testing;
+
+using iox::roudi::IceOryxRouDiMemoryManager;
+
+namespace iox
+{
+namespace test
+{
+
+/// @brief This test file verifies that the BaseClass IceoryxRouDiMemoryManager is tested
+class IceoryxRoudiMemoryManager_test : public Test
+{
+  public:
+    std::unique_ptr<IceOryxRouDiMemoryManager> m_roudiMemoryManagerTest;
+
+    void SetUp() override
+    {
+      auto config = iox::RouDiConfig_t().setDefaults();
+      m_roudiMemoryManagerTest = std::unique_ptr<IceOryxRouDiMemoryManager>(new IceOryxRouDiMemoryManager(config));
+    }
+
+    void TearDown() override
+    {
+    }
+};
+
+TEST_F(IceoryxRoudiMemoryManager_test, ConstructorSuccess)
+{
+    EXPECT_THAT(m_roudiMemoryManagerTest, Not(Eq(nullptr)));
+}
+
+TEST_F(IceoryxRoudiMemoryManager_test, IntrospectionMemoryManagerNulloptWhenNotPresent)
+{
+    auto result = m_roudiMemoryManagerTest->introspectionMemoryManager();
+    EXPECT_THAT(result, Eq(iox::cxx::nullopt_t()));
+}
+
+TEST_F(IceoryxRoudiMemoryManager_test, segmentManagerNulloptWhenNotPresent)
+{
+    auto resultTest = m_roudiMemoryManagerTest->segmentManager();
+    EXPECT_THAT(resultTest, Eq(iox::cxx::nullopt_t()));
+}
+
+TEST_F(IceoryxRoudiMemoryManager_test, portPoolNulloptWhenNotPresent)
+{
+    auto testResult = m_roudiMemoryManagerTest->portPool();
+    EXPECT_THAT(testResult, Eq(iox::cxx::nullopt_t()));
+}
+
+TEST_F(IceoryxRoudiMemoryManager_test, CreateAndAnnouceMemoryHasNoError)
+{
+    auto testResult = m_roudiMemoryManagerTest->createAndAnnounceMemory();
+    EXPECT_THAT(testResult.has_error(), Eq(false));
+}
+
+TEST_F(IceoryxRoudiMemoryManager_test, MgmtMemoryProviderReturnNonNullPtr)
+{
+    auto testResult = m_roudiMemoryManagerTest->mgmtMemoryProvider();
+    EXPECT_THAT(testResult, Not(Eq(nullptr)));
+}
+
+TEST_F(IceoryxRoudiMemoryManager_test, AcquiringIntrospectionMemoryManagerAfterCreateAndAnnounceMemoryIsSuccessful)
+{
+    auto tr = m_roudiMemoryManagerTest->createAndAnnounceMemory();
+
+    EXPECT_THAT(tr.has_error(), Eq(false));
```

Comment:
@JimmyBch for the future as a side note: Everything you already tested you can assume as working and doesn't need to be tested again - except when it would cause a segfault when it is not working.

In this case you have already tested `CreateAndAnnouceMemoryHasNoError` therefore you do not need to test it here again.

But for now it is fine - you do not need to adjust it.